docs/tasks/manage-stateful-set/upgrade-pet-set-to-stateful-set.md

```diff
@@ -0,0 +1,152 @@
+This page shows how to upgrade from Pet Sets (Kubernetes cluster at version 1.3-1.4) to Stateful Sets (version >= 1.5).
+{% endcapture %}
+
+{% capture prerequisites %}
+
+* If you don't have Pet Sets in your current cluster, or you don't plan to upgrade your cluster/master to >= 1.5 at this moment, you can skip this task now. 
+
+{% endcapture %}
+
+{% capture steps %}
+
+### Differences between alpha Pet Sets and beta Stateful Sets
+
+Pet Set was introduced as an alpha resource in Kubernetes release 1.3, and was renamed to Stateful Set as a beta resource in 1.5. 
+Here are some notable changes:
+
+* **Stateful Set is the new Pet Set**: Pet Set is no longer available in any Kubernetes release >= 1.5. It becomes beta Stateful Set. To know why the name is changed, see this [discussion thread](https://github.com/kubernetes/kubernetes/issues/27430).
+* **Stateful Set guards against split brain**: Stateful Sets guarantee at most one pod for a given ordinal index can be running anywhere in a cluster, in order to guard against split brain scenarios with distributed applications. *TODO: Link to doc about fencing*
+* **Flipped debug annotation behavior**: The default value of the debug annotation (`pod.alpha.kubernetes.io/initialized`) is now `true`. The absence of this annotation will pause Pet Set operations, but will NOT pause Stateful Set operations. In most cases, you no longer need this annotation in your Stateful Set manifests. 
+
+
+### Upgrading from Pet Sets to Stateful Sets
+
+Note that these steps need to be operated in the specified order. You **should
+NOT upgrade your Kubernetes master, nodes, or `kubectl` to >= 1.5 version yet**,
+until told to do so.
+
+#### Find all Pet Sets and their manifests 
+
+First, find all existing Pet Sets in your cluster:
+
+```shell
+kubectl get petsets --all-namespaces
+```
+
+If you don't find any existing Pet Sets, you can safely upgrade your cluster to >= 1.5.
+
+If you find existing Pet Sets and you have all their manifests at hand, you can continue to the next step to prepare Stateful Set manifests. 
+
+Otherwise, you need to save their manifests so that you can recreate them as Stateful Sets later. 
+Here's an example command for you to save all existing Pet Sets as one file. 
+
+```shell
+# Save all existing Pet Sets in all namespaces into a single file. Only needed when you don't have their manifests at hand. 
+kubectl get petsets --all-namespaces -o yaml > all-petsets.yaml
+```
+
+#### Prepare Stateful Set manifests 
+
+Now, for every Pet Set manifest you have, prepare a corresponding Stateful Set manifest with the following steps: 
+
+1. Change `apiVersion` from `apps/v1alpha1` to `apps/v1beta1`
+2. Change `kind` from `PetSet` to `StatefulSet`
+3. If you have the debug hook annotation `pod.alpha.kubernetes.io/initialized` set to `true`, you may remove it since it's redundant. If you don't have this annotation, you should add one, with the value set to `false`, to pause Stateful Sets operations.
```

Comment:
Suggestion: you can remove it because

Generally, we use these conventions:
may: permission
might: possibility
can: able to
because: why
since: from some time in the past